src/main/java/com/google/jenkins/plugins/storage/StorageUtil.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2013 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jenkins.plugins.storage;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.jenkins.plugins.storage.AbstractUploadDescriptor.GCS_SCHEME;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import com.google.jenkins.plugins.credentials.domains.RequiresDomain;
```

Comment:
I've fixed the imports in the newly introduced files. mvn checkstyle gives many errors, a lot of them pre-existing.
I think it would be cleaner in the history if I go through them in a follow-up PR